### PR TITLE
fix for new version

### LINKED ISSUE #2: MissingMethodException
Hello,
if you die while having the Artifact installed you'll get an MissingMethodException which results in an endless game ending screen.
Also if any enemy dies the console will be spammed by MissingMethodExceptions.

### PR DESCRIPTION
**Pull request by SirHamburger**

### CODE REVIEW

**Comment by NotTsunami** [comment]
Hey, never got a notification for this so it kinda slipped by me. Sorry for the delays and thank you for your contribution! Looking into this now.

File: ArtifactOfOrder/ArtifactOfOrder.cs [hunk 1]
```diff
@@ -3,23 +3,35 @@
...
         {
             Order.nameToken = "Artifact of Order";
             Order.descriptionToken = "Applies the sequencing effect of a Shrine of Order to a player upon death.";
-            Order.smallIconDeselectedSprite = CreateSprite(Properties.Resources.artifact_unselected);
-            Order.smallIconSelectedSprite = CreateSprite(Properties.Resources.artifact_selected);
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream("ArtifactOfOrder.artifactoforder"))
+            {
+                var bundle = AssetBundle.LoadFromStream(stream);
+                Order.smallIconSelectedSprite = bundle.LoadAsset<Sprite>("Assets/Import/Icons/artifact_selected.png");
+                Order.smallIconDeselectedSprite = bundle.LoadAsset<Sprite>("Assets/Import/Icons/artifact_unselected.png");
```
Comment (NotTsunami) [inline ArtifactOfOrder/ArtifactOfOrder.cs:30]: Okay, I'm fine with switching to an AssetBundle if it simplifies some code.

File: ArtifactOfOrder/ArtifactOfOrder.csproj [hunk 2]
```diff
@@ -6,69 +6,70 @@
...
     <Reference Include="UnityEngine.ImageConversionModule">
-      <HintPath>..\..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\Risk of Rain 2\Risk of Rain 2_Data\Managed\UnityEngine.ImageConversionModule.dll</HintPath>
+      <HintPath>./libs/UnityEngine.ImageConversionModule.dll</HintPath>
     </Reference>
     <Reference Include="UnityEngine.Networking">
-      <HintPath>..\..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\Risk of Rain 2\Risk of Rain 2_Data\Managed\UnityEngine.Networking.dll</HintPath>
+      <HintPath>./libs/UnityEngine.Networking.dll</HintPath>
+    </Reference>
+    <Reference Include="UnityEngine.AssetBundleModule">
+      <HintPath>/home/sirhamburger/Git/ArtifactOfDoom/libs/UnityEngine.AssetBundleModule.dll</HintPath>
```
Comment (NotTsunami) [inline ArtifactOfOrder/ArtifactOfOrder.csproj]: This is a hardcoded path to your `AssetBundleModule.dll`. Please correct this to the included dll.

File: ArtifactOfOrder/ArtifactOfOrder.cs [hunk 3]
```diff
@@ -3,23 +3,35 @@
 using R2API.Utils;
 using RoR2;
 using UnityEngine;
+using System.Reflection;
+using R2API;
```
Comment (NotTsunami) [inline ArtifactOfOrder/ArtifactOfOrder.cs]: Please sort these.

**Review by NotTsunami — CHANGES_REQUESTED** [review]

File: ArtifactOfOrder/ArtifactOfOrder.cs (hunk 1 above)
Comment (SirHamburger) [reply ArtifactOfOrder/ArtifactOfOrder.cs:30]: The change to an AssetBundle makes the whole Resources.Designer.cs Class obsolete. So i think the code is much easier to understand. 
But the main reason for the change was that i wasn't able to fix it without it.

File: ArtifactOfOrder/ArtifactOfOrder.csproj (hunk 2 above)
Comment (SirHamburger) [reply ArtifactOfOrder/ArtifactOfOrder.csproj]: I did change it with the new commit. Also i removed unused Includes.

File: ArtifactOfOrder/ArtifactOfOrder.cs (hunk 3 above)
Comment (SirHamburger) [reply ArtifactOfOrder/ArtifactOfOrder.cs]: Hopefully they should be sorted now.

**Comment by SirHamburger** [comment]
With the latest commit i added a Try Catch around the GetDioCount. So orig(self, body); will still get called if there is an error in GetDioCount.

**Comment by NotTsunami** [comment]
Thanks a bunch! I'm gonna run this later today and merge it if there are no issues.

**Comment by NotTsunami** [comment]
Thank you!

**Pull request merged**